### PR TITLE
Fix notification for new conversations

### LINKED ISSUE #5330: Private messages do not make notification for new message when new thread is open
Steps to reproduce : send a new private message to someone (not a new answer to a previous PM), he will receive no notification.

### PR DESCRIPTION
**Pull request by svbergerem**
Fixes #5330, maybe introduced by [this commit](https://github.com/diaspora/diaspora/commit/33c3b38f2f8c6ede39b1357eb3cb49f18543cdce). The spec I posted in #5330 wasn't working but I wrote a cucumber feature.

### CODE REVIEW

**Comment by Flaburgan** [comment]
@svbergerem hm, if the commit you linked is the problem, maybe we have other regression elsewhere.

**Comment by jhass** [comment]
Mmh, you mean that self.attributes creates something that makes it always unique or never matching?

**Comment by svbergerem** [comment]
@jhass Here is an example of `self.attributes`:
`{"id"=>nil, "subject"=>"test9", "guid"=>"30cbe7809a5e0132c4c9047d7b8eb256", "author_id"=>1, "created_at"=>Thu, 19 Feb 2015 12:10:17 UTC +00:00, "updated_at"=>nil}`

I guess there is no object with the same attributes in the db so we try to create a new one which fails because there is already a conversation with the same guid in the db.

File: app/models/conversation.rb [hunk 1]
```diff
@@ -86,7 +86,7 @@ def subscribers(user)
   end
 
   def receive(user, person)
-    cnv = Conversation.find_or_create_by!(self.attributes)
+    cnv = Conversation.create_with(self.attributes).find_or_create_by!(self.attributes[:guid])
```
Comment (jhass) [inline app/models/conversation.rb]: Doesn't this have to be .`.find_or_create_by!(guid: guid)` then? This just seems to insert a `WHERE 123456abcdef` into the query if I'm following correctly.

Comment (svbergerem) [inline app/models/conversation.rb]: Already tried that. In that case we create a second conversation in the cukes which makes them fail:

Comment (jhass) [inline app/models/conversation.rb]: Just checked, it generates the following query:

```sql
SELECT  "conversations".* FROM "conversations"  LIMIT 1
```

That just happens to match in the test.

Comment (jhass) [inline app/models/conversation.rb]: `Conversation.create_with(attributes).find_or_create_by!(guid: guid)` generates: `SELECT  "conversations".* FROM "conversations"  WHERE "conversations"."guid" = 'cf7cd2e09a6001329fdd14feb5c40cd3' LIMIT 1` and doesn't fail the test for me.

**Comment by svbergerem** [comment]
@Flaburgan Of course we have other regressions somewhere else but I don't think that the linked commit caused them. :-P

All other changes in the commit seem legit.

**Comment by jhass** [comment]
Alright, good catch! Thank you!

**Comment by Flaburgan** [comment]
Thank you Steffen! This bug was so annoying!

**Comment by marienfressinaud** [comment]
I strongly agree, thanks a lot!

**Pull request merged**